tensorflow/python/data/experimental/ops/distribute.py

```diff
@@ -49,22 +47,13 @@ def __init__(self, input_dataset, num_workers, index):
     self._input_dataset = input_dataset
 
     self._element_spec = input_dataset.element_spec
-    if (compat.forward_compatible(2019, 11, 25) or
```

Comment:
You can remove the `compat.forward_compatible` part but you should probably leave both sides of the `if` statement here as there is the other condition on the `auto_shard_policy` setting.

Reply:
`compat.forward_compatible(2019, 11, 25)` will always evaluate to `True`, so the second condition will never be called  since a logic like `if True or auto_shard_policy_condition` will always be `True` regardless of the `auto_shard_policy_condition`.